src/widget/wtrackmenu.cpp

```diff
@@ -535,8 +535,12 @@ void WTrackMenu::setupActions() {
         }
     }
 
-    if (featureIsEnabled(Feature::RemoveFromDisk) &&
-            m_pTrackModel->hasCapabilities(TrackModel::Capability::RemoveFromDisk)) {
+    if (featureIsEnabled(Feature::RemoveFromDisk)) {
+        if (m_pTrackModel &&
+                !m_pTrackModel->hasCapabilities(
+                        TrackModel::Capability::RemoveFromDisk)) {
+            return;
+        }
```

Comment:
I hope I got that right...

Reply:
You don't want an early return in this case. Also `m_pTrackModel != nullptr` is not checked for any of the other options. It should be checked only once earlier if needed.

What was wrong with the existing code?

Reply:
My assumption when looking at how the track menu is initialized for either `WTrackproperty`, `WTrackWidgetGroup` or `WTrackTableView` is that we can end up here only if either there's a track model or a track loaded. So no track model = track loaded, right?
So the existing code would add the action to the track widgets' menu.

:facepalm: oouh I'm so shortsighted at this time of the day.. of course I don't want to quit `setupActions()`
will fix

Reply:
> What was wrong with the existing code?

Ha :bulb: I didn't question my own code you already reviewed ; ]
`featureIsEnabled()` already checks `m_pTrackModel->hasCapabilities()` (if the feature requires a track model) so that was redundant in the first place...